gateway-service/src/main/java/org/zowe/apiml/gateway/conformance/ValidateAPIController.java

```diff
@@ -10,96 +10,203 @@
+    private static final String InvalidServiceIdRegex = "[^a-z0-9]";
+
+
+    private static final String wrongServiceIdKey = "org.zowe.apiml.gateway.verifier.wrongServiceId";
+    private static final String NoMetadataKey = "org.zowe.apiml.gateway.verifier.noMetadata";
+    private static final String NonConformantKey = "org.zowe.apiml.gateway.verifier.nonConformant";
+
+
+    private static final String problemWithRegistration = "Registration problems";
+    private static final String problemWithMetadata = "Metadata problems";
+    private static final String problemWithConformance = "Conformance problems";
+
+
     private final MessageService messageService;
 
     private final VerificationOnboardService verificationOnboardService;
-    private String invalidKey = "org.zowe.apiml.gateway.verifier.wrongServiceId";
+
+    private final DiscoveryClient discoveryClient;
+
 
     /**
-     * Accept serviceID and return the JSON file with appropirate message to show if it is valid
-     *
-     * @param serviceID accepted serviceID to check validation
-     * @return return the JSON file message of whether the serviceID is valid
+     * Accepts serviceID and checks conformance criteria
      *
+     * @param serviceId accepted serviceID to check for conformance
+     * @return 200 if service is conformant, 400 + JSON explanation if not
      */
-    @PostMapping(
-        value = "/validate",
+    @GetMapping(
+        value = "/gateway/conformance/{serviceId}",
         produces = MediaType.APPLICATION_JSON_VALUE
     )
-    public ResponseEntity<Object> checkValidate(@RequestBody String serviceID) {
+    @HystrixCommand
+    public ResponseEntity<String> checkConformance(@PathVariable String serviceId) {
+
+
+        ConformanceProblemsContainer foundNonConformanceIssues = new ConformanceProblemsContainer();
+
+
+        foundNonConformanceIssues.put(problemWithConformance, validateServiceIdFormat(serviceId));
 
-        String message = validator(serviceID);
 
-        if (!message.isEmpty()) {
-            return new ResponseEntity<>(messageService.createMessage(invalidKey, message).mapToApiMessage(), HttpStatus.BAD_REQUEST);
+        if (foundNonConformanceIssues.size() != 0) {
+            return GenerateBadRequestResponseEntity(NonConformantKey, foundNonConformanceIssues);
         }
 
-        return new ResponseEntity<>(HttpStatus.OK);
+        foundNonConformanceIssues.put(problemWithRegistration, checkOnboarding(serviceId));
+
+        if (foundNonConformanceIssues.size() != 0) {     // cant continue if a service isn't registered
+            return GenerateBadRequestResponseEntity(wrongServiceIdKey, foundNonConformanceIssues);
+        }
+
+        List<ServiceInstance> serviceInstances = discoveryClient.getInstances(serviceId);
+        foundNonConformanceIssues.put(problemWithRegistration, instanceCheck(serviceInstances));
+
+        ServiceInstance serviceInstance = serviceInstances.get(0);
```

Comment:
This can cause IndexOutOfBoundException.

Reply:
Okay, I see that instanceCheck() validates if serviceInstances is not empty. So in theory there should not be any exception.